Tools/cases_generator/generate_cases.py

```diff
@@ -1153,25 +1206,22 @@ def write_pseudo_instrs(self) -> None:
             self.out.emit(f"    ((OP) == {op}) || \\")
         self.out.emit(f"    0")
 
-    def emit_metadata_entry(self, name: str, fmt: str, flags: int) -> None:
-        flags_strs = [f"{name}{INSTR_FLAG_SUFFIX}"
-                      for i, name in enumerate(INSTRUCTION_FLAGS) if (flags & (1<<i))]
-        flags_s = "0" if not flags_strs else ' | '.join(flags_strs)
+    def emit_metadata_entry(self, name: str, fmt: str, flags: InstructionFlags) -> None:
         self.out.emit(
-            f"    [{name}] = {{ true, {INSTR_FMT_PREFIX}{fmt}, {flags_s} }},"
+            f"    [{name}] = {{ true, {INSTR_FMT_PREFIX}{fmt}, {flags} }},"
```

Comment:
Honestly, overloading `str()` to mean "whatever we want in C" feels a little too clever. Maybe just have and call a method designed to return that specific string?